packages/react-core/src/modules/in-app-message/hooks/useInAppMessages.ts

```diff
@@ -29,37 +35,9 @@ export const useInAppMessages = <TContent = GenericData, TData = GenericData>(
     return new InAppMessagesClient(inAppChannelClient, messageType, options);
   }, [inAppChannelClient, messageType, options]);
 
-  // TODO: Create selectors as functions in the client library though
-  const messages = useStore(inAppChannelClient.store, (state) => {
-    const messageIds = new Set(
-      state.queries[inAppMessagesClient.queryKey]?.data?.entries?.map(
-        (message) => message.id,
-      ),
-    );
-
-    // TODO: Just grab based on the message id instead of looping over all messages
-    const messages = Object.entries(state.messages).reduce<
-      InAppMessage<TContent, TData>[]
-    >((messages, [id, message]) => {
-      if (messageIds.has(id)) {
-        messages.push(message as InAppMessage<TContent, TData>);
-      }
-      return messages;
-    }, []);
-
-    return messages;
-  });
-
-  // TODO: Consolidate to a single useStore call to get all message info
-  const { networkStatus, loading } = useStore(
+  const { messages, networkStatus, loading } = useStore(
     inAppChannelClient.store,
-    (state) => {
-      const query = state.queries[inAppMessagesClient.queryKey];
-      return {
-        networkStatus: query?.networkStatus ?? NetworkStatus.ready,
-        loading: query?.loading ?? false,
-      };
-    },
+    inAppMessagesClient.getQueryInfoSelector<TContent, TData>,
```

Comment:
I'm wondering out loud if this helper really belongs to `inAppMessagesClient` i.e. is it really `inAppMessagesClient` concern? 

Asking because from what I can tell, this selector returns an object that is relevant to the `useInAppMessage` hook, and not something you care if you are using `inAppMessagesClient` hook directly without this hook? If so, then this selector should probably live in this file?

Reply:
I placed it in the message client so that it can be used outside of React. If someone were to use the message client with say Vue, they would need to use this same selector to get query info for the given message + filters. We can't fully abstract the fact that we're using Tanstack store because for the store to be reactive, it needs to be accessed using the correct `useStore` hook for the given framework. Given that, I think providing helper methods like this is a good solution so that non-React users only have to do something like:

```
// my-component.vue
<script setup>
import { useStore } from '@tanstack/vue-store';
import { store } from './store';

const { ...message info } = useStore(store, (state) => inAppMessageClient.getQueryInfoSelector(store))
</script>
```

Reply:
Ah okay I see. 

Just to double check, this query info data is only set by the `fetch()` call, right? Meaning, the "query"  in `getQueryInfoSelector()` helper refers to `fetch()`, and the selector would return empty data if you haven't called `fetch()`?

Reply:
Yeah, it's only set by fetch currently. The realtime connection will also update the same info when a new message is received (e.g. the list of messages associated with a set of filters/message type)